source/array_table_model.ts

```
@@ -55,32 +58,82 @@ export class ArrayTableModel extends TableModel {
    * @throws RangeError - The source or destination are not within this table's
    *                      range.
    */
-  public moveRow(source: number, destination: number): void {}
+  public moveRow(source: number, destination: number): void {
+    if(source > this.rowCount || source < 0) {
```

If ```source === this.rowCount``` then you should also throw an exception.